DOC: Add Hugging Face Hub access

lhoestq opened the pull request:
Hi team ! We recently made most of the 200k+ datasets from the community on Hugging Face loadable in pandas :)
And we even show the code snippet to load a dataset on each dataset page (see images below)

In this PR I added documentation on accessing Hugging Face Datasets using `fsspec`. It uses `hf://` paths and the implementation is part of the `huggingface-hub` library. Let me know what you think !


1. "Use this dataset" button on a dataset page:
<img width="1290" alt="image" src="https://github.com/user-attachments/assets/6ab57ddf-c42d-4892-8572-6bd0e772e7ce" />

2. Modal showing [pandas code snippet](https://huggingface.co/datasets/google/FACTS-grounding-public?library=pandas):
<img width="1294" alt="image" src="https://github.com/user-attachments/assets/c08ca12f-3b98-4748-8569-8d71892234c3" />

closes https://github.com/pandas-dev/pandas/issues/46000

lhoestq committed:
Update pyproject.toml

lhoestq committed:
Update install.rst

lhoestq committed:
Update io.rst

julien-c reviewed (approved):
very useful IMO!

mroeschke reviewed (changes requested):
Thanks for the PR, but based on the discussion in https://github.com/pandas-dev/pandas/issues/46000, we would want to include this package in the `ecosystem.md` docs and not have a pip extra.

lhoestq committed:
remove pip extra

lhoestq committed:
Update ecosystem.md

lhoestq committed:
link to docs

lhoestq commented:
Added a section in ecosystem.md :) And I removed the pip extra, but I think the HF fsspec implementation should be an option at the same level as S3 or GCS no ?

I kept HF as an option in the docs on fsspec support in `io.rst` as well, let me know if it's fine.

rhshadrach commented on doc/source/user_guide/io.rst:
```diff
@@ -1565,7 +1565,7 @@ of header key value mappings to the ``storage_options`` keyword argument as show
 All URLs which are not local files or HTTP(s) are handled by
 `fsspec`_, if installed, and its various filesystem implementations
-(including Amazon S3, Google Cloud, SSH, FTP, webHDFS...).
+(including Amazon S3, Google Cloud, Hugging Face, SSH, FTP, webHDFS...).
```
This is meant to give examples and not be an exhaustive list. If we leak ecosystem entries into various parts of the docs, it seems to me we are at great risk of leaving behind stale docs as the ecosystem evolves over time. I'd prefer to remove this.

lhoestq replied:
I see, I reverted the change. Let me know if there is a better place to list the main fsspec implementations

lhoestq committed:
Revert change in io.rst

mroeschke commented:
Thanks @lhoestq
